### PR TITLE
INTMDB-526-Feature federated settings role mapping

### PR DESCRIPTION
**Pull request by govardhanpagidi**
## Description

Please include a summary of the fix/feature/change, including any relevant motivation and context.

Link to any related issue(s): 

## Type of change:

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] This change requires a documentation update

## Required Checklist:

- [x] I have signed the [MongoDB CLA](https://www.mongodb.com/legal/contributor-agreement)
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] I have added any necessary documentation (if appropriate)
- [x] I have run `make fmt` and formatted my code

### CODE REVIEW

File: cfn-resources/federated-settings-org-role-mapping/Makefile [hunk 1]
```diff
@@ -0,0 +1,11 @@
+.PHONY: build test clean
+
+build:
+	make -f makebuild  # this runs build steps required by the cfn cli
+
+test:
```
Comment (rdharmedran) [inline cfn-resources/federated-settings-org-role-mapping/Makefile:6]: This file is not matching with make file of  other modules .
please compare it with cluster module

File: cfn-resources/federated-settings-org-role-mapping/cmd/resource/resource.go [hunk 2]
```diff
@@ -0,0 +1,343 @@
...
+			Message:          err.Error(),
+			OperationStatus:  handler.Failed,
+		}, nil
+	}
+
+	federationSettingsID := currentModel.FederationSettingsId
+	orgID := currentModel.OrgId
+
+	if (currentModel.ExternalGroupName) == nil {
+		err := errors.New("error creating federated settings org role mapping: ExternalGroupName should be set when `Export` is set")
```
Comment (rdharmedran) [inline cfn-resources/federated-settings-org-role-mapping/cmd/resource/resource.go:59]: do we need to check this?.We are already validating "ExternalGroupName" through validator module.

var CreateRequiredFields = []string{"FederationSettingsId", "OrgId", "**ExternalGroupName**", "RoleAssignments", "ApiKeys.PrivateKey", "ApiKeys.PublicKey"}

File: cfn-resources/federated-settings-org-role-mapping/cmd/resource/resource.go [hunk 3]
```diff
@@ -0,0 +1,343 @@
...
+		_, _ = log.Warnf("Create - error: %+v", err)
+		return handler.ProgressEvent{
+			OperationStatus:  handler.Failed,
+			Message:          err.Error(),
+			HandlerErrorCode: cloudformation.HandlerErrorCodeInvalidRequest}, nil
+	}
+	if (currentModel.RoleAssignments) == nil || len(currentModel.RoleAssignments) == 0 {
+		err := errors.New("error creating federated settings org role mapping: RoleAssignments should be set when `Export` is set")
```
Comment (rdharmedran) [inline cfn-resources/federated-settings-org-role-mapping/cmd/resource/resource.go:67]: we can avoid creating error here .

File: cfn-resources/federated-settings-org-role-mapping/cmd/resource/resource.go [hunk 4]
```diff
@@ -0,0 +1,343 @@
...
+	federationSettingsID := currentModel.FederationSettingsId
+	orgID := currentModel.OrgId
+	roleMappingID := currentModel.Id
+
+	// Check if  already exist
+	if !isRoleMappingExists(currentModel, client) {
+		return progressevents.GetFailedEventByCode("Not Found", cloudformation.HandlerErrorCodeNotFound), nil
+	}
+	if (currentModel.RoleAssignments) == nil || len(currentModel.RoleAssignments) == 0 {
+		err := errors.New("error creating federated settings org role mapping: RoleAssignments should be set when `Export` is set")
```
Comment (rdharmedran) [inline cfn-resources/federated-settings-org-role-mapping/cmd/resource/resource.go:163]: avoid this.Use string constant

File: cfn-resources/federated-settings-org-role-mapping/README.md [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+cfn invoke DELETE test.request.json 
+```
+
+Both UPDATE & DELETE tests must pass.
+
+## Installation
+TAGS=logging make
+cfn submit --verbose --set-default
+
+## Cloudformation Examples
```
Comment (Zuhairahmed) [inline cfn-resources/federated-settings-org-role-mapping/README.md:36]:
```suggestion
## CloudFormation Example
```

File: cfn-resources/federated-settings-org-role-mapping/README.md [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+
+Please see the [CFN Template](test/federated-setting-org-role-mapping.sample-template.json) for example resource
```
Comment (Zuhairahmed) [inline cfn-resources/federated-settings-org-role-mapping/README.md:38]: Broken link

File: cfn-resources/federated-settings-org-role-mapping/README.md [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+And then you can create the stack with a helper script it insert the apikeys for you:
+
+
+```bash
+repo_root=$(git rev-parse --show-toplevel)
+source <(${repo_root}/quickstart-mongodb-atlas/scripts/export-mongocli-config.py)
+${repo_root}/quickstart-mongodb-atlas/scripts/launch-quickstart.sh ${repo_root}/cfn-resources/federated-setting-org-role-mapping/test/federated-setting-org-role-mapping.sample-template.json SampleAccessList1 ParameterKey=FederationSettingsId,ParameterValue=<Federation-Settings-Id> ParameterKey=OrgId,ParameterValue=<Connected-Organization-Id>
+```
+
+For more information see: MongoDB Atlas API [Federated Setting Org Role Mapping Endpoint](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Federated-Authentication) Documentation.
```
Comment (Zuhairahmed) [inline cfn-resources/federated-settings-org-role-mapping/README.md:64]:
```suggestion
For more information see: MongoDB Atlas API Endpoint [Federated Setting Org Role Mapping](https://www.mongodb.com/docs/atlas/reference/api-resources-spec/#tag/Federated-Authentication) Documentation.
```

**Review by Zuhairahmed — APPROVED** [review]
Few minor comments, not blocking. LGTM! Thanks @govardhanpagidi!

File: cfn-resources/federated-settings-org-role-mapping/cmd/resource/resource.go (hunk 4 above)
Comment (govardhanpagidi) [reply cfn-resources/federated-settings-org-role-mapping/cmd/resource/resource.go:147]: Fixed

File: cfn-resources/federated-settings-org-role-mapping/Makefile (hunk 1 above)
Comment (govardhanpagidi) [reply cfn-resources/federated-settings-org-role-mapping/Makefile]: Fixed

**Review by ParthasarathyVaradhan — APPROVED** [review]
LGTM. Build file for multiregion deployment is straight forward.

**Review by swattyT — APPROVED** [review]

**Pull request merged**